### PR TITLE
[Doc] improve `thread::Thread` and `thread::Builder` documentations

### PR DESCRIPTION
**Pull request by gamazeps**
Part of #29378

- Adds information about the stack_size when using `Builder`. This might be considered too low level, but I assume that if someone wants to create their own builder instead of using `thread::spawn` they may be interested in that info.
- Updates the `thread::Thread` structure doc, mostly by explaining how to get one, the previous example was removed because it was not related to `thread::Thread`, but rather to `thread::Builder::name`.
  Not much is present there, mostly because this API is not often used (the only method that seems useful is `unpark`, which is documented in #41809).

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -244,6 +244,11 @@ impl Builder {
     /// Generates the base configuration for spawning a thread, from which
     /// configuration methods can be chained.
     ///
+    /// If the [`stack_size`][stack_size] field is not specified, the stack size
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: changing this to just have 

    [`stack_size`]

would follow convention

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -259,6 +264,8 @@ impl Builder {
     ///
     /// handler.join().unwrap();
     /// ```
+    ///
+    /// [stack_size]: ../../std/thread/struct.Builder.html#method.stack_size
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: that means writing

    [`stack_size`]: ../../std/


here

File: src/libstd/thread/mod.rs [hunk 3]
```diff
@@ -714,33 +721,21 @@ struct Inner {
 
 #[derive(Clone)]
 #[stable(feature = "rust1", since = "1.0.0")]
-/// A handle to a thread.
+/// Threads are represented via the `Thread` type, which you can get in one of
+/// two ways:
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: This first line needs to be a summary; this doesn't quite work. I think leaving the first line, adding a `///` in between, and keeping this new stuff is 👍

**Review by steveklabnik — COMMENTED** [review]
some small details! thank you!

**Comment by gamazeps** [comment]
Done :)

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks a ton!

**Pull request merged**